mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraCombinationManagerTest.java

```diff
@@ -44,12 +44,8 @@
 import org.apache.james.mailbox.store.quota.NoQuotaManager;
 import org.junit.After;
 import org.junit.Before;
-import org.junit.ClassRule;
 
 public class CassandraCombinationManagerTest extends AbstractCombinationManagerTest {
-
-    @ClassRule public static DockerCassandraRule cassandraServer = new DockerCassandraRule();
```

Comment:
Why don't you put the Singleton use into the DockerCassandraRule ? That would help encapsulate these kind of changes for the better.

Reply:
Because we might not wnt to use the singleton when we use the DockerCassandraRule

Exemple: Cassandra JMAP integration tests.

BTW having a rule that does nothing (because it relies on a singleton) is basically a lie and is IMO harmfull. Why not expose clearly the reality (that we rely on a singleton) in the test classes?

Reply:
Let's argue : 

1. the tests classes don't care about how many cassandra we run, is it a docker container or not, etc, it just requires a running cassandra
2. having a Rule is not perfect, we could use an extension instead, but in both cases the intent is to encapsulate a behavior concerning a dependency for the test
3. for now, the singleton may make the Rule obsolete but changing only the Rule/Extension would allow encapsulating the cassandra provider policy, I think that it worth keeping it for easy changes in the future

Reply:
Ok, I'll try to propose something like this.

Reply:
Done. Note that a separate singleton is necessary to avoid weird static order stuff...